### PR TITLE
Connects Scholarship List view to scholarship page

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
When connected to a functioning backend, the scholarship part works.

### CODE REVIEW

**Review by hzimz — APPROVED** [review]

**Review by MikolJiron — APPROVED** [review]

File: src/main/webapp/js/datahandlers/singlepagedatahandler.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ class SinglePageDataHandler {
    * @return {*} - The JSON response.
    */
   async fetchJson_(id) {
-    const response = await fetch(this.getRequestPath_(), {'id': id });
+    const response = await fetch(`${this.getRequestPath_()}?id=${id}`);
```
Comment (tinagrasso) [inline src/main/webapp/js/datahandlers/singlepagedatahandler.js]: Doesn't getRequestPath take an id as an argument?

File: src/main/webapp/js/finscholar/appstate/appstate.js [hunk 2]
```diff
@@ -51,12 +55,39 @@ class AppState {
...
+    const id = node.id;
+    if (node.classList.contains('colleges')) {
+      this.currentView_ = new CollegePageView();
+    } else {
+      this.currentView_ = new ScholarshipPageView();
+    }
+    this.currentView_.setId(id);
+    await this.currentView_.renderView();
+    this.navbarInstance_ = new NavBar();
+    this.navbarInstance_.registerListener(this.navbarUpdate.bind(this));
```
Comment (tinagrasso) [inline src/main/webapp/js/finscholar/appstate/appstate.js]: This is identical to below. Abstract away into its own method.

File: src/main/webapp/js/finscholar/viewfactory/viewfactory.js [hunk 3]
```diff
@@ -39,9 +38,6 @@ const navbarViewFactory = (index) => {
     case 2:
       view = new ScholarshipListView();
       break;
-    case 3:
-      view = new ScholarshipPageView();
-      break;
```
Comment (tinagrasso) [inline src/main/webapp/js/finscholar/viewfactory/viewfactory.js:44]: Why are you removing this and why are there 2 cases for ScholarshipListView?

**Review by tinagrasso — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/datahandlers/singlepagedatahandler.js (hunk 1 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/datahandlers/singlepagedatahandler.js]: Oh yeah... I added id to a branch basing off from this branch. Just changed it

File: src/main/webapp/js/finscholar/viewfactory/viewfactory.js (hunk 3 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/finscholar/viewfactory/viewfactory.js:44]: College list view is not implemented in this branch. I just left case 1 blank

File: src/main/webapp/js/finscholar/appstate/appstate.js (hunk 2 above)
Comment (Lorraine-Lyu) [reply src/main/webapp/js/finscholar/appstate/appstate.js]: done

File: src/main/webapp/js/finscholar/viewfactory/viewfactory.js (hunk 3 above)
Comment (tinagrasso) [reply src/main/webapp/js/finscholar/viewfactory/viewfactory.js:44]: Ok cool. In the future, don't add implementation for things that don't exist. Leave the case blank with just the break; statement and a comment explaining that the view doesn't exist yet.

**Review by tinagrasso — APPROVED** [review]

**Review by CarlosM10011 — APPROVED** [review]
We need to fix the compiler warnings.

**Pull request merged**